agent/consul/gateways/bind.go

```diff
@@ -71,3 +84,52 @@ func getReferences(route structs.BoundRoute) (referenceSet, gatewayRefs) {
 
 	return parentRefs, gatewayRefs
 }
+
+func requestToResourceRef(req controller.Request) structs.ResourceReference {
+	ref := structs.ResourceReference{
+		Kind: req.Kind,
+		Name: req.Name,
+	}
+	if req.Meta != nil {
+		ref.EnterpriseMeta = *req.Meta
+	}
+	return ref
+}
+
+// RemoveGateway sets the route status for the given gateway to be unbound if it should be bound
+func RemoveGateway(gateway structs.ResourceReference, entries ...structs.BoundRoute) []structs.ControlledConfigEntry {
+	now := time.Now()
+	modified := []structs.ControlledConfigEntry{}
+	for _, route := range entries {
+		updater := structs.NewStatusUpdater(route)
+		for _, parent := range route.GetParents() {
+			if parent.Kind == gateway.Kind && parent.Name == parent.Name && parent.EnterpriseMeta.IsSame(&gateway.EnterpriseMeta) {
+				updater.SetCondition(structs.Condition{
+					Type:               "Bound",
+					Status:             "False",
+					Reason:             "GatewayNotFound",
+					Message:            "gateway was not found",
+					Resource:           &parent,
+					LastTransitionTime: &now,
+				})
+			}
+		}
+		if toUpdate, shouldUpdate := updater.UpdateEntry(); shouldUpdate {
+			modified = append(modified, toUpdate)
+		}
+	}
+	return modified
+}
+
+// RemoveRoute unbinds the route from the given gateways, returning the list of gateways that were modified.
+func RemoveRoute(route structs.ResourceReference, entries ...*gatewayMeta) []*gatewayMeta {
```

Comment:
This is triggered when a route gets deleted by using the route itself as a resource ref -- see the comment above on the switch of `unbindRoute` to use refs instead of `structs.BoundRoute`.